src/Compilers/CSharp/Portable/Symbols/MemberSignatureComparer.cs

```diff
@@ -526,7 +585,63 @@ public bool Equals(Symbol? member1, Symbol? member2)
                 }
             }
 
-            return !_considerTypeConstraints || HaveSameConstraints(member1, typeMap1, member2, typeMap2);
+            return !_considerTypeConstraints || HaveSameConstraints(member1, typeMap1, member2, typeMap2, ignoreNullability: (_typeComparison & TypeCompareKind.IgnoreNullableModifiersForReferenceTypes) != 0);
+
+            static bool haveSameParameterNames(ImmutableArray<ParameterSymbol> params1, ImmutableArray<ParameterSymbol> params2)
+            {
+                return params1.SequenceEqual(params2, (p1, p2) => p1.Name == p2.Name);
+            }
+
+            static bool haveSameTypeParameterNames(ImmutableArray<TypeParameterSymbol> typeParams1, ImmutableArray<TypeParameterSymbol> typeParams2)
+            {
+                return typeParams1.SequenceEqual(typeParams2, (p1, p2) => p1.Name == p2.Name);
+            }
+
+            static bool haveSameParameterAttributes(ImmutableArray<ParameterSymbol> params1, ImmutableArray<ParameterSymbol> params2)
+            {
+                return params1.SequenceEqual(params2, (p1, p2) => hasSameAttribute(p1.GetAttributes(), p2.GetAttributes()));
+            }
+
+            static bool haveSameTypeParameterAttributes(ImmutableArray<TypeParameterSymbol> typeParams1, ImmutableArray<TypeParameterSymbol> typeParams2)
+            {
+                return typeParams1.SequenceEqual(typeParams2, (p1, p2) => hasSameAttribute(p1.GetAttributes(), p2.GetAttributes()));
+            }
+
+            static bool hasSameAttribute(ImmutableArray<CSharpAttributeData> attributes1, ImmutableArray<CSharpAttributeData> attributes2)
+            {
+                if (attributes1.Length != attributes2.Length)
+                {
+                    return false;
+                }
+
+                var comparer = CommonAttributeDataComparer.InstanceIgnoringNamedArgumentOrder;
```

Comment:
>InstanceIgnoringNamedArgumentOrder [](http://example.com/codeflow?start=59&length=34)

This makes sense for the extensions scenario, but is not obvious from the API perspective. It might be good to add a comment for ```considerAttributes``` parameter.